libraries/botbuilder-ai/src/luisRecognizerOptionsV3.ts

```diff
@@ -69,56 +69,73 @@ export class LuisRecognizerV3 extends LuisRecognizerInternal {
-                                                entityName: key,
-                                                startIndex: start,
-                                                entityLength: end - start,
-                                                resolution: value,
-                                            });
-                                        }
-                                    });
-                                }
-                                return externalEntities;
-                            }, options.externalEntities);
+    async recognizeInternal(context: DialogContext | TurnContext): Promise<RecognizerResult>;
+
+    /**
+     * Calls LUIS to recognize intents and entities in a users utterance.
+     *
+     * @param {string} utterance The utterance to be recognized.
+     */
+    async recognizeInternal(utterance: string): Promise<RecognizerResult>;
+
+    /**
+     * @internal
+     */
+    async recognizeInternal(contextOrUtterance: DialogContext | TurnContext | string): Promise<RecognizerResult> {
+        if (typeof contextOrUtterance === 'string') {
+            const utterance = contextOrUtterance;
+            return this.recognize(null, utterance, this.predictionOptions);
+        } else {
+            if (contextOrUtterance instanceof DialogContext) {
+                const dialogContext = contextOrUtterance;
+                const activity = dialogContext.context.activity;
+                let options = this.predictionOptions;
+                if (options.externalEntityRecognizer) {
+                    // call external entity recognizer
+                    const matches = await options.externalEntityRecognizer.recognize(dialogContext, activity);
+                    // TODO: checking for 'text' because we get an extra non-real entity from the text recognizers
+                    if (matches.entities && Object.entries(matches.entities).length) {
+                        options = {
+                            apiVersion: 'v3',
+                            externalEntities: [],
+                        };
+                        const entities = matches.entities;
+                        const instance = entities['$instance'];
+                        if (instance) {
+                            Object.entries(entities)
+                                .filter(([key, _value]) => key !== 'text' && key !== '$instance')
+                                .reduce((externalEntities: ExternalEntity[], [key, value]) => {
+                                    const instances: unknown[] = instance[`${key}`];
+                                    const values: unknown[] = Array.isArray(value) ? value : [];
+                                    if (instances?.length === values?.length) {
+                                        instances.forEach((childInstance) => {
+                                            if (
+                                                childInstance &&
```

Comment:
Here is potentially a good place for `runtypes`:

```typescript
if (t.Record({startIndex: t.Number, endIndex: t.Number}).guard(childInstance)) {
  childInstance.startIndex + childInstance.endIndex;
}
```

Reply:
[Resolved](https://github.com/microsoft/botbuilder-js/pull/3690/files#diff-c3651a7a67ef5f2bb9b7e930d218e0a8d9976572855bc60bfb0016c4ee061e7eR113), assuming `childInstance.startIndex + childInstance.endIndex;` was only an example and not meant to be used.